File: Python_Codes/7. Exception Handling/CustomException.py
# Define a custom exception
class InvalidAgeException(Exception):
    def __init__(self, message):
        super().__init__(message)


# Function to validate age
def validate_age(age):
    if age < 18:
        raise InvalidAgeException(f"Age {age} is not valid. Must be 18 or above.")
    print("Age is valid:", age)


def main():
    try:
        # Test the custom exception by providing an invalid age
        validate_age(16)
    except InvalidAgeException as e:
        print("Caught Custom Exception:", e)


if __name__ == "__main__":
    main()

File: Projects/JavaProject/src/com/Exception_Handling/CustomException.java
package com.Exception_Handling;

class InvalidAgeException extends Exception {
    public InvalidAgeException(String message) {
        super(message);
    }
}

public class CustomException {

    // The method validates the age and throws our custom exception if necessary
    public static void validateAge(int age) throws InvalidAgeException {
        if (age < 18) {
            throw new InvalidAgeException("Age " + age + " is not valid. Must be 18 or above.");
        }
        System.out.println("Age is valid: " + age);
    }

    public static void main(String[] args) {
        try {
            // Test the custom exception by providing an invalid age
            validateAge(16);
        } 
        catch (InvalidAgeException e) {
            System.out.println("Caught Custom Exception: " + e.getMessage());
        }
    }
}
    


